### ISSUE TITLE
Text does not wrap automatically

### ISSUE THREAD

**Issue by bharathy-kannan** [ShaMan123/react-native-math-view#22]
Hi,

I am trying to embed math equations in between text. I am unable to do this as the font size automatically changes and all the text gets printed in one line making the question unreadable. I am not sure where I am going wrong. Here is an example of the same.
In the first line, I have written this within the MathView tag

'\text{This text does not wrap. This text does not wrap. This text does not wrap} \dfrac{1}{2}'
'\text{This text does not wrap} \dfrac{1}{2}' 

![image](https://user-images.githubusercontent.com/53509255/72163085-26d34d00-33e9-11ea-8c59-78aec1560ba8.png)

**Comment by Kings-dev-02** [comment]
Hi Kannan,

Output is generated as svg so it will not get wrapped, instead of that split the text and formula and send only formaula to mathview

Ex:   <View style={{
            flexDirection: 'row',
            flexWrap: "wrap",
          }}>
            <Text>{"This text does not wrap. This text does not wrap. This text does not wrap"}</Text>
            <MathView
               math = '\dfrac{1}{2}''
            /><View>

       ios below will work, in android you have to give width and height in styles for math view
        <Text>
            <Text>{"This text does not wrap. This text does not wrap. This text does not wrap"}</Text>
            <MathView
                math = '\dfrac{1}{2}''
            /><Text>

**Comment by ShaMan123** [comment]
Thanks @Kings-dev-02 
@bharathy-kannan follow @Kings-dev-02 suggestion.
Android shouldn't require width and height in style. Does it?

**Comment by bharathy-kannan** [comment]
> Hi Kannan,
> 
> Output is generated as svg so it will not get wrapped, instead of that split the text and formula and send only formula to mathview
> 
> Ex: <View style={{
> flexDirection: 'row',
> flexWrap: "wrap",
> }}>
> {"This text does not wrap. This text does not wrap. This text does not wrap"}
> <MathView
> math = '\dfrac{1}{2}''
> />
> 
> ```
>    ios below will work, in android you have to give width and height in styles for math view
>     <Text>
>         <Text>{"This text does not wrap. This text does not wrap. This text does not wrap"}</Text>
>         <MathView
>             math = '\dfrac{1}{2}''
>         /><Text>
> ```
Hi,
What if I have a question like this?
It is given that \Delta ABC \sim \Delta PQR with \frac{BC}{QR} = \frac{1}{3}. Then, \frac{ar(\Delta ABC)}{ar)\Delta PQR)} is equal to: 
In this case, I would have to parse it several times. Is there any other option? 

Thanks for the response. Appreciate it.

**Comment by ShaMan123** [comment]
Read #21 - maybe webview? This library has been created to eliminate the use of webview due to performance and styling issues. However, wrap content is valuable, so maybe the trade off is worth while. Give it a try.
Besides that... As mentioned above, break it down to bits and render separately.

**Comment by bharathy-kannan** [comment]
I tried using react-native-katex library which is a webview but then looks like here too it parses only expressions and not the normal text. I really like MathView. Is there any parser that you can suggest? Also looks like the Latex delimiters $ ... $ does not seem to work with MathView.

**Comment by ShaMan123** [comment]
create an example so I can have a look

**Comment by bharathy-kannan** [comment]
I have just used the Katex render option and have tried out an expression. Again it does not parse normal text. The text comes in italics. If I place them in \text mode, then it does not wrap. How do I use normal text inside Katex?

function getContent({ inlineStyle, expression = '', ...options }) {
  const tryOutExpression = 'It is given that \\Delta ABC \\sim \\Delta PQR with \\frac{BC}{QR} = \\frac{1}{3}. Then, \\frac{ar(\\Delta ABC)}{ar)\\Delta PQR)} is equal to: '  
  return `<!DOCTYPE html>

  <html>
<head>
<style>
${katexStyle}
${inlineStyle}
</style>
<script>
window.onerror = e => document.write(e);
window.onload = () => katex.render(${JSON.stringify(tryOutExpression)}, document.body, ${JSON.stringify(options)});
${katexScript}
</script>
</head>
<body>
</body>
</html>
`;
}

const defaultStyle = StyleSheet.create({
  root: {
    height: 20,
  },
  questionText:{
    fontSize: 20,
    color:'black',
    justifyContent: 'center',
    alignItems:'center',
    alignSelf:'center',
    padding:10,
  },
});

const defaultInlineStyle = `
html, body {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0;
  padding: 0;
}
.katex {
  margin: 0;
  display: flex;  
  font-size:3em;
}
`;

export default class Katex extends Component {

  static propTypes = {
    expression: string.isRequired,
    displayMode: bool,
    throwOnError: bool,
    errorColor: string,
    inlineStyle: string,
    macros: object,
    output:string,
    trust:bool,
    colorIsTextColor: bool,
    onLoad: func,
    onError: func,
  };

  static defaultProps = {
    expression: '',
    displayMode: false,
    throwOnError: false,
    errorColor: '#f00',
    inlineStyle: defaultInlineStyle,
    style: defaultStyle,
    macros: {},
    output: 'htmlAndMathml',
    trus:true,
    colorIsTextColor: false,
    onLoad: () => {},
    onError: () => {},
  };

  render() {
    const { style, onLoad, onError, ...options } = this.props;
 
    return (
    <View style={{flex:1}}>
      <WebView
      style={style}
      source={{ html: getContent(options) }}
      onLoad={onLoad}
      onError={onError}
      renderError={onError}
    />
  
    </View>
    )
  }
}
![image](https://user-images.githubusercontent.com/53509255/72202044-e1faf500-3480-11ea-88c8-b05989a4a5d4.png)

Really appreciate your help

**Comment by ShaMan123** [comment]
Please create a gist or sandbox etc.

**Comment by ShaMan123** [comment]
what if you replace $ with $$?

**Comment by ShaMan123** [comment]
https://www.mathjax.org/#demo

**Comment by bharathy-kannan** [comment]
I have used Katex. May be MathJax would work. In Katex component that I have taken, it does not understand $ or $$. But what do others who use MathView usually do? Do they create their own mark up and parse to display the text properly?

**Comment by ShaMan123** [comment]
Try this :

It is given that $\Delta ABC \sim \Delta PQR$ with $\frac{BC}{QR} = \frac{1}{3}$. Then, $\frac{ar(\Delta ABC)}{ar)\Delta PQR)}$ is equal to

**Comment by ShaMan123** [comment]
![image](https://user-images.githubusercontent.com/34343793/72202357-12ce3080-3467-11ea-96aa-017b750ffeac.png)

**Comment by ShaMan123** [comment]
This should work with MathView as well

**Comment by bharathy-kannan** [comment]
It works with MathJax Sandbox that you have shared. But unfortunately, if I paste this in MathView RN component, does not work. 

The text part is not getting wrapped and they are all trying to fit in a single line. Difficult to read without zooming.
Here is what I did
        <View style={styles.container}>
          <View style={{
flexDirection: 'row',
flexWrap: "wrap",flex:1
}}>
      <MathView math='It is given that $\Delta ABC \sim \Delta PQR$ with $\frac{BC}{QR} = \frac{1}{3}$. Then, $\frac{ar(\Delta ABC)}{ar)\Delta PQR)}$ is equal to' />
</View>

![image](https://user-images.githubusercontent.com/53509255/72202394-f3de9700-3484-11ea-9bef-c20d0b4d2759.png)

Appreciate your patience

**Comment by ShaMan123** [comment]
Without a proper example I find it hard to help.
So create your own sandbox or better off, create a react native app on a new git repository and share it with me.

**Comment by bharathy-kannan** [comment]
Sure. I will upload in git and share it with you

**Comment by bharathy-kannan** [comment]
Hi,
I have created the repo here
https://github.com/bharathy-kannan/mathviewexample

Bascially I want to use normal text along with Math equations embedded inbetween. The questions will come from a database dynamically. The code can be seen in App.js where I have used both MathView and Katex to show a sample

Here is the screen shot of the code that has been uploaded
![image](https://user-images.githubusercontent.com/53509255/72204713-01a21580-34a1-11ea-8e4e-77479ce55231.png)

Appreciate your help.

**Comment by Kings-dev-02** [comment]
> Thanks @Kings-dev-02
> @bharathy-kannan follow @Kings-dev-02 suggestion.
> Android shouldn't require width and height in style. Does it?

Ya below is not working in android it's asking for height and width if I use it as inline text, any help would be much appreciated

 <Text>
        <Text>{"This text does not wrap. This text does not wrap. This text does not wrap"}</Text>
        <MathView
            math = '\dfrac{1}{2}''
        /><Text>

**Comment by Kings-dev-02** [comment]
<Text>
        <Text>{"This text does not wrap. This text does not wrap. This text does not wrap"}</Text>
        <MathView
            math = '\dfrac{1}{2}''
        /><Text>

**Comment by Kings-dev-02** [comment]
> Hi,
> I have created the repo here
> https://github.com/bharathy-kannan/mathviewexample
> 
> Basically I want to use normal text along with Math equations embedded inbetween. The questions will come from a database dynamically. The code can be seen in App.js where I have used both MathView and Katex to show a sample
> 
> Here is the screen shot of the code that has been uploaded
> ![image](https://user-images.githubusercontent.com/53509255/72204713-01a21580-34a1-11ea-8e4e-77479ce55231.png)
> 
> Appreciate your help.

Kannan,

I dont know whats your source of input, i assume you may take html input and want to render the same with math expression in mobile view, if so easy way is while creating html content wrap it inside your custom tag,
 Ex: Here is a normal text with some math in it using <mymath data='\frac{1}{2}'> \frac{1}{2} </mymath>

above will still work in the browser without any problem your custom tag will be ignored and you can use mathjax or katex to render, when it comes to react-native, parse that tag replace with mathview

Thanks,
Kingslee

**Comment by bharathy-kannan** [comment]
Hi,

I plan to just store plain text with Latex strings in the database without Html. Thanks for the suggestion though. It seems like a good idea. 
Any idea why Katex in webview too does not render? Is there something wrong with the code
that I have written?
 
Thanks
Bharathy Kannan

**Comment by ShaMan123** [comment]
I sent a PR.
apparently text should be wrapped as follows:
```
math={'\\text{my text }\\frac{1}{2}'}
```
mathjax docs are unkind.

**Comment by ShaMan123** [comment]
@Kings-dev-02 regarding setting width/height did you try following the example's components?

**Comment by bharathy-kannan** [comment]
If I use <Text /> and <MathView /> separately, I dont have to give height or width. But if I embed MathView inside Text component, it expects height and width. I guess I will have to go with the first option as I may not know the height and width of the rendered element. But I would request you to sincerely consider having the normal mode with math mode as a feature request. This will help many

**Comment by ShaMan123** [comment]
You shouldn't wrap MathView with Text.
I have added another screen to the example app: [Inline Text](https://github.com/ShaMan123/react-native-math-view/blob/master/MathExample/src/WrapWithText.tsx).
It accepts line breaking and $ $$.
You should both take a look.
I might merge this as default behavior in the future.

**Comment by bharathy-kannan** [comment]
I resolved the problem somehow. However, I am not sure that the <Text> component and <MathView> do not gel together and look distinct when written with flexWrap. Any suggestions here? Sometimes they come on different lines too. @ShaMan123 should I take a pull again? Also how to make the font size of normal text and MathView text look alike? Any suggestions on the font face?

Here is my code 

`<View style={{flexDirection:'row', 
          flexWrap:"wrap", paddingLeft:10, alignItems:'center', paddingBottom:40}}>
          <Text style={styles.questionText}>If ABC and DEF are similar triangles such that </Text>
          <MathView  style={styles.questionText} math={'\\angle A=47\\ degree'} />
          <Text  style={styles.questionText}>and</Text> 
          <MathView  style={styles.questionText} math={'\\angle E=83\\ degree'} />
          <Text  style={styles.questionText}>then</Text>
          <MathView  style={styles.questionText} math={'\\angle C = ?'} />          
          </View>`

**Comment by ShaMan123** [comment]
Man, I'm having a hard time understanding what you mean.
Upload some screenshots, perhaps.
Revise your PR, definitely.
**And dive into the example app!!**
A lot can be learned from the snippets there which I deliberately wrote.
